src/frontend/src/pages/TeamsPage/DescriptionPageBlock.tsx

```diff
@@ -0,0 +1,123 @@
+import { Team } from 'shared';
+import { Edit } from '@mui/icons-material';
+import LoadingIndicator from '../../components/LoadingIndicator';
+import ErrorPage from '../ErrorPage';
+import PageBlock from '../../layouts/PageBlock';
+import ReactMarkdown from 'react-markdown';
+import styles from '../../stylesheets/pages/teams.module.css';
+
+interface DescriptionPageBlockProps {
+  team: Team;
+}
+
+const DescriptionPageBlock: React.FC<DescriptionPageBlockProps> = ({ team }) => {
+  const auth = useAuth();
+  const theme = useTheme();
+  const [isEditingDescription, setIsEditingDescription] = useState(false);
+  const [description, setDescription] = useState(team.description);
+  const [isPreview, setIsPreview] = useState(false);
+  const { isLoading, isError, error, mutateAsync } = useEditTeamDescription(team.teamId);
+
+  if (isError) return <ErrorPage message={error?.message} />;
+  if (isLoading) return <LoadingIndicator />;
+
+  const handleSubmit = async () => {
+    await mutateAsync(description);
+    setIsEditingDescription(false);
+  };
+
+  const hasPerms = () => {
+    return !(
+      auth.user &&
+      (auth.user.role === 'ADMIN' || auth.user.role === 'APP_ADMIN' || auth.user.userId === team.leader.userId)
+    );
+  };
+
+  const countWords = (str: string): number => {
+    const words = str.split(' ');
+    let wordCount = 0;
+    words.forEach((word) => {
+      if (word.trim() !== '') {
+        wordCount++;
+      }
+    });
+    return wordCount;
+  };
+
+  const checkWordCount = (str: string, limit: number): boolean => {
+    return countWords(str) < limit;
+  };
+
+  const headerRight = (
+    <div style={{ display: 'flex' }}>
+      <Button onClick={() => setIsEditingDescription(false)}>Cancel</Button>
+      <Button
+        onClick={() => setIsPreview(!isPreview)}
+        sx={{
+          backgroundColor: theme.palette.grey[600],
+          color: theme.palette.getContrastText(theme.palette.grey[600]),
+          '&:hover': {
+            backgroundColor: theme.palette.grey[700]
+          }
+        }}
+      >
+        Preview
+      </Button>
+      <Button
+        sx={{
+          ml: 2,
+          backgroundColor: theme.palette.success.main,
+          color: theme.palette.success.contrastText,
+          '&:hover': {
+            backgroundColor: theme.palette.success.dark
+          }
+        }}
+        onClick={handleSubmit}
+      >
+        Save
+      </Button>
+    </div>
+  );
+
+  const editingView = (
+    <PageBlock title={'Description'} headerRight={headerRight}>
+      {isPreview ? (
+        <ReactMarkdown className={styles.markdown}>{description}</ReactMarkdown>
+      ) : (
+        <TextField
+          fullWidth
+          multiline
+          rows={10}
+          value={description}
+          onChange={(e) => setDescription(e.target.value)}
+          inputProps={{
+            maxlength: checkWordCount(description, 300) ? null : 0
```

Comment:
I think this looks like a fine way to do it. You might check and see if the helper text can be turned red when the count goes too high, which I think is achievable with feeling the helper text an isError Boolean or something like that (see other usages of helper text). 

The only other ways I might think to do word limit is to check on submit before executing the API request, but this seems better. Also technically im pretty sure we should be checking word count in the back-end too to be proper and thorough.

Reply:
I think we need to do both because people will be able to copy and paste words into the text field and if its over the limit it wont throw an error unless you check it.

Reply:
But in regards to back end support I’m wondering if we want to check it in the service, or in the route validator, if thats possible.